src/modules/Jobs/jobCreate/containers/JobCreateForm/useJobCreateForm.tsx

```diff
@@ -278,16 +256,16 @@ export const useJobCreateForm = () => {
     };
 
     try {
-      const res = isEdit ? await updateJob(jobDetail?.id, jobPayload) : await createJob(jobPayload);
-      questions.forEach(async (q: QuestionReq) => {
+      const res = isEdit && jobDetail?.id ? await updateJob(jobDetail.id, jobPayload) : await createJob(jobPayload);
+      questions.forEach(async (q: Question) => {
         if (q?.id) {
           await updateQuestionJob(q?.project_id, q?.id, {
             question: q.question,
             required: q.required,
-            options: q.options,
+            options: q.options || undefined,
           });
         } else {
-          await addQuestionJob(res.id, q);
+          await addQuestionJob(res.id, q as QuestionReq);
```

Comment:
I can't understand why `q` is `Question` in some places and `QuestionReq` in other places.
Please rename `q` to `question`

Reply:
Because the same state and form is used for creating and updating questions. Creating questions needs Question type (without id), but updating needs (QuestionReq) with id. That's why we have cast them to each other in several points